trademapper/js/trademapper.imageexport.js

```diff
@@ -0,0 +1,43 @@
+define([
+	"jquery"
+],
+
+function($) {
+	return {
+		/**
+		 * Constructor
+		 *
+     * This creates a hidden link which allows a download to be emulated
+		 * by settings its href attribute to a dataURL extracted from the map SVG.
+		 *
+		 * button: element with an on() method which can be used to bind clicks
+		 * svgElement: DOM node; reference to the <svg> element containing the map
+		 */
+		init: function(button, svgElement) {
+			// clicks on the button export the SVG
+			button.on("click", this.exportSvg.bind(this));
+
+			this.svgElement = $(svgElement);
```

Comment:
In trademapper.js, `this.mapRootElement`, `this.tmsvg`, `this.zomg` etc. are all [d3 selection objects]( https://github.com/d3/d3-selection). It's a bit confusing that here `this.svgElement` is a jQuery object.

Three options I can think of:

1. Leave as is, this file is self contained so there is no direct clash;
2. Use a '$' prefix for variables that contain jQuery objects, eg. `this.$svgElement`;
3. Store the DomElement, not the jQuery object and use `$` when needed.

I prefer 3 over 1. I often use 2, though I don't know if it would be consistent with the rest of the code base. And it might not be to everyone's taste.

Putting this here, but I leave it up to you to make the final decision :)

Reply:
I'll use the '$' prefix. I don't really like this, but it is consistent with other parts of the code base.